src/memo_list.js

```diff
@@ -0,0 +1,46 @@
+import "./memo_list.css";
+import PropTypes from "prop-types";
+import React from "react";
+
+export default function MemoList({
+  memos,
+  addMemo,
+  selectedId,
+  setSelectedId,
+}) {
+  const memoList = Object.entries(memos).map(([key, value]) => {
+    const head = value.split("\n")[0];
+    if (key === selectedId) {
+      return (
+        <button className="memo-list-button selected" key={key}>
+          {head}
+        </button>
+      );
+    } else {
+      return (
+        <button
+          className="memo-list-button"
+          key={key}
+          onClick={() => setSelectedId(key)}
+        >
+          {head}
+        </button>
+      );
+    }
```

Comment:
こんな具合でも良い気がしますね！
onClick は選択済みの場合でも setSelectedId をしても良いかと思いますー（同じ値がセットされるので再レンダリングされないはず）

```suggestion
  <button
    className={`memo-list-button ${key === selectedId ? 'selected' : ''}`}
    key={key}
    onClick={() => setSelectedId(key)}
  >
    {head}
  </button>
```